src/app/core/state/session/session.store.ts

```diff
@@ -0,0 +1,32 @@
+import { Injectable } from '@angular/core';
+import { Store, StoreConfig } from '@datorama/akita';
+import { Session } from './session.interface';
+
+export interface SessionState {
+  session: Session;
+}
+
+export function createInitialState(): SessionState {
+  return {
+    session: {
+      accessToken: undefined,
+      tokenType: undefined,
+    },
+  };
+}
+
+@Injectable({ providedIn: 'root' })
+@StoreConfig({ name: 'session' })
+export class SessionStore extends Store<SessionState> {
+  constructor() {
+    super(createInitialState());
+  }
+
+  login(session: Session) {
```

Comment:
I don't like the idea of adding more methods in the Store. I saw that this is what Datorama suggested in their Authentication article, but it feels that it can easily get confusing over time.

I recommend handling the logic in the Service and only using the store to update the state. 

So
`session.service.ts` would call
```
this.sessionStore.update(...
```

Thoughts?

Reply:
I agree, the code has been moved to the akita service